4MD/src/main/java/com/infoshareacademy/four_md/controller/RecipeController.java

```diff
@@ -46,20 +53,29 @@ public String saveRecipe(@RequestBody @Valid @ModelAttribute Recipe recipe, Bind
         if (bindResult.hasErrors()) {
             return "add-recipe";
         }
-        dbRecipeRepository.save(recipe);
+
+        User currentUser = getCurrentUser();
+        currentUser.getListOfRecipes().add(recipe);
+        userProvider.save(currentUser);
+
         model.addAttribute("recipeName", recipe.getName());
         return "confirmation";
     }
 
     @GetMapping("/delete/{id}")
     public String deleteRecipe(@PathVariable int id) throws IOException {
-        dbRecipeRepository.remove(id);
+        User currentUser = getCurrentUser();
+
+        currentUser.getListOfRecipes().removeIf(s -> s.getId() == id);
+        userProvider.save(currentUser);
+
+
         return "redirect:/recipes/list";
     }
 
     @GetMapping("/edit/{id}")
-    public String editRecipe(@PathVariable int id, Model model) throws IOException {
-        Recipe recipe = dbRecipeRepository.get(id);
+    public String editRecipe(@PathVariable int id, Model model) {
+        Recipe recipe = getCurrentUser().getListOfRecipes().get(id);
```

Comment:
Recipe recipe = getCurrentUser().getListOfRecipes().get(id); wyskakuje błąd index out of band. Tak jakby id były przechowywane w tablicy zaczynajacej sie od 0. 
Recipe recipe = getCurrentUser().getListOfRecipes().get(id-1); dopiero wtedy pobierze dobry przepis